### ISSUE TITLE
Permission issue on install on empty machine high sierra

### ISSUE THREAD

**Issue by daften** [geerlingguy/ansible-role-homebrew#82]
When I use this role on a freshly reset machine (high sierra), I get the error from #76  and it seems to be because the /usr/local/var folder doesn't exist and brew update doesn't seem to create that folder.

Output from my run and some manual steps after that on the file:
[output.txt](https://github.com/geerlingguy/ansible-role-homebrew/files/1395743/output.txt)

After that output there's another non-existing folder that is apparently needed.

**Comment by daften** [comment]
Just reset a laptop again and confirmed this

**Comment by Anima-t3d** [comment]
Still happening on fresh install OSX 10.13.1 (High Sierra)

**Comment by elliotweiser** [comment]
I'm going to cut a High Sierra packer box so I can start testing these roles against the new OS.

**Comment by jameshiew** [comment]
Doing a sudo command before running the role seems to be a workaround - run something like `sudo echo test` then run the role/playbook straight after from the same shell.

**Comment by kas21** [comment]
I am just starting to use this role and am encountering similar issues.  Is there a reason that /usr/local/{Cellar, Frameworks, bin, etc, include, lib, opt, sbin, share, var} aren't created during the homebrew install?  The ruby script that is used for a manual install creates these folders and sets owner and group permissions when executed.  I don't see that occurring in this role.

**Comment by holms** [comment]
I've just installed new high sierra setup and it's the same, and it seems to be quite a broad issue from what I've googled around from stackoverflow.

```
Romans-iMac:roles holms$ brew update
mkdir: /usr/local/var/homebrew/locks: Permission denied
Error: Can't create update lock in /usr/local/var/homebrew/locks!
Fix permissions by running:
  sudo chown -R $(whoami) /usr/local/var/homebrew
Romans-iMac:roles holms$   sudo chown -R $(whoami) /usr/local/var/homebrew
chown: /usr/local/var/homebrew: No such file or directory
```

**Comment by daften** [comment]
@holms That seems to be because you installed homebrew as root, which is not intended. Different problem than this issue, where the role doesn't create some necessary folders.

**Comment by daften** [comment]
That error: `Error: Can't create update lock in /usr/local/var/homebrew/locks!` is a permission issue AFTER installing homebrew, in your case, during a brew update. This issue is about the homebrew installation. Your issue is caused most of the time because people installing homebrew as root, which made me jump to that conclusion.
That PR also sets the permissions on some folders correctly, in your case /usr/local/var, that's why it works for you. Ansible does this because it runs as root (i'm guessing you use ansible-playbook -K) and it sets the owner to the user running ansible with owner: "{{ ansible_user_id }}".

TL;DR; stil the wrong issue for what you experienced :)

**Comment by holms** [comment]
Interesting, all I've been doing is running ansible like so:

```
ansible-playbook main.yml -i inventory -K
```

I've been using this role to install brew, never used anything else, it's absolutely fresh machine in here, I've just checked out repository and installed ansible through pip - that's it.

**Comment by elliotweiser** [comment]
@daften Bugfix merged in #88 and released in 2.3.3

**Comment by daften** [comment]
Awesome, thanks!
